controller/rancher/rancher_cert_dir_test.go

```diff
@@ -87,6 +87,22 @@ func TestCheckCertDirUpdate(t *testing.T) {
 		t.Fatalf("Error building config %v", err)
 	}
 
+	if len(newConfigs[0].Certs) != 3 {
+		t.Fatalf("Failed to read the Updated certificates from the directory")
+	}
+
+	err = ioutil.WriteFile("./testcerts/certs/c.com/privkey.pem", d1, 0644)
+	if err != nil {
+		t.Fatalf("Error writing file %v", err)
+	}
+
+	time.Sleep(10 * time.Second)
```

Comment:
sleep 10 is just to make sure the next poll has run in the meanwhile. The poll interval is set to 5 seconds at the init() of the test.